JSQCoreDataKit/JSQCoreDataKit/CoreDataExtensions.swift

```diff
@@ -146,47 +146,61 @@ public func deleteObjects <T: NSManagedObject>(objects: [T], inContext context:
 }
 
 /**
+ Asynchronously resets a `CoreDataStack` instance.
  Resets the managed object contexts in the stack, then deletes and recreates the persistent store connected to the coordinator.
  For binary and SQLite stores, this will also remove the store from disk.
 
- - parameter stack: The `CoreDataStack` instance to be reset.
+ - note: This operation is performed on a background queue.
 
- - throws: If the reset fails or errors, then this function throws an `NSError`.
+ - parameter stack:      The `CoreDataStack` instance to be reset.
+ - parameter queue:      A background queue on which to reset the stack. The default is a high priority background queue.
+ - parameter completion: The closure to be called once resetting is complete.
  */
-public func resetStack(stack: CoreDataStack) throws {
-    stack.mainContext.performBlockAndWait {
-        stack.mainContext.reset()
-    }
+public func resetStackInBackground(
+    stack: CoreDataStack,
+    queue: dispatch_queue_t = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0),
+    completion: CoreDataStackFactory.CompletionHandler) {
```

Comment:
Not sure if `CompletionHandler` should be taken out of `CoreDataStackFactory` into `CoreDataExtensions`, any thoughts @jessesquires ?